### ISSUE TITLE
lora微调过程中卡住不动

### ISSUE THREAD

**Issue by ljhjxt** [OpenBMB/MiniCPM-V#638]
### 是否已有关于该错误的issue或讨论？ | Is there an existing issue / discussion for this?

- [x] 我已经搜索过已有的issues和讨论 | I have searched the existing issues / discussions

### 该问题是否在FAQ中有解答？ | Is there an existing answer for this in FAQ?

- [X] 我已经搜索过FAQ | I have searched FAQ

### 当前行为 | Current Behavior

lora微调时候从第三个step开始卡住不动，一张卡可以，但是八张卡就不行
Parameter Offload: Total persistent parameters: 3039200 in 759 params
  0%|                                                                                                                            | 1/2500 [00:11<7:50:22, 11.29s/it]tried to get lr value before scheduler/optimizer started stepping, returning lr=0
tried to get lr value before scheduler/optimizer started stepping, returning lr=0
tried to get lr value before scheduler/optimizer started stepping, returning lr=0
tried to get lr value before scheduler/optimizer started stepping, returning lr=0
tried to get lr value before scheduler/optimizer started stepping, returning lr=0
tried to get lr value before scheduler/optimizer started stepping, returning lr=0
tried to get lr value before scheduler/optimizer started stepping, returning lr=0
tried to get lr value before scheduler/optimizer started stepping, returning lr=0
{'loss': 1.1679, 'grad_norm': 0.0, 'learning_rate': 0, 'epoch': 0.0}                                                                                                
  0%|                                                                                                                            | 2/2500 [00:21<7:21:39, 10.61s/it]tried to get lr value before scheduler/optimizer started stepping, returning lr=0
tried to get lr value before scheduler/optimizer started stepping, returning lr=0
tried to get lr value before scheduler/optimizer started stepping, returning lr=0
tried to get lr value before scheduler/optimizer started stepping, returning lr=0
tried to get lr value before scheduler/optimizer started stepping, returning lr=0
tried to get lr value before scheduler/optimizer started stepping, returning lr=0
tried to get lr value before scheduler/optimizer started stepping, returning lr=0
tried to get lr value before scheduler/optimizer started stepping, returning lr=0
{'loss': 0.0639, 'grad_norm': 0.0, 'learning_rate': 0, 'epoch': 0.0}                                                                                                
  0%|▏                                                                                                                           | 3/2500 [00:27<5:49:44,  8.40s/it]tried to get lr value before scheduler/optimizer started stepping, returning lr=0
tried to get lr value before scheduler/optimizer started stepping, returning lr=0
tried to get lr value before scheduler/optimizer started stepping, returning lr=0
tried to get lr value before scheduler/optimizer started stepping, returning lr=0
tried to get lr value before scheduler/optimizer started stepping, returning lr=0
tried to get lr value before scheduler/optimizer started stepping, returning lr=0
tried to get lr value before scheduler/optimizer started stepping, returning lr=0
tried to get lr value before scheduler/optimizer started stepping, returning lr=0
{'loss': 0.0639, 'grad_norm': 0.0, 'learning_rate': 0, 'epoch': 0.01}                                                                                               
  0%|▏                                                                                                                           | 3/2500 [00:27<5:49:44,  8.40s/it]

以下是参数配置
GPUS_PER_NODE=8
NNODES=1
NODE_RANK=0
MASTER_ADDR=localhost
MASTER_PORT=6001

MODEL="/root/paddlejob/workspace/linjianhong/movie_comment/minicpm/model/models--openbmb--MiniCPM-V-2_6"

DATA="/root/paddlejob/workspace/linjianhong/movie_comment/minicpm/code/MiniCPM-V/finetune/finetune_train_timechat_data.json"
EVAL_DATA="/root/paddlejob/workspace/linjianhong/movie_comment/minicpm/code/MiniCPM-V/finetune/finetune_val_timechat_data.json"
LLM_TYPE="qwen2" 

MODEL_MAX_Length=4096 # if conduct multi-images sft, please set MODEL_MAX_Length=4096

DISTRIBUTED_ARGS="
    --nproc_per_node $GPUS_PER_NODE \
    --nnodes $NNODES \
    --node_rank $NODE_RANK \
    --master_addr $MASTER_ADDR \
    --master_port $MASTER_PORT
"
torchrun $DISTRIBUTED_ARGS finetune.py  \
    --model_name_or_path $MODEL \
    --llm_type $LLM_TYPE \
    --data_path $DATA \
    --eval_data_path $EVAL_DATA \
    --remove_unused_columns false \
    --label_names "labels" \
    --prediction_loss_only false \
    --bf16 false \
    --bf16_full_eval false \
    --fp16 true \
    --fp16_full_eval true \
    --do_train \
    --do_eval \
    --tune_vision true \
    --tune_llm false \
    --use_lora true \
    --lora_target_modules "llm\..*layers\.\d+\.self_attn\.(q_proj|k_proj|v_proj|o_proj)" \
    --model_max_length $MODEL_MAX_Length \
    --max_slice_nums 1 \
    --max_steps 2500 \
    --eval_steps 250 \
    --output_dir output/output_lora \
    --logging_dir output/log_lora \
    --logging_strategy "steps" \
    --per_device_train_batch_size 1 \
    --per_device_eval_batch_size 1 \
    --gradient_accumulation_steps 1 \
    --evaluation_strategy "steps" \
    --save_strategy "steps" \
    --save_steps 250 \
    --save_total_limit 10 \
    --learning_rate 1e-6 \
    --weight_decay 0.1 \
    --adam_beta2 0.95 \
    --warmup_ratio 0.01 \
    --lr_scheduler_type "cosine" \
    --logging_steps 1 \
    --gradient_checkpointing true \
    --deepspeed zero3_offload.json \
    --report_to "tensorboard" # wandb
